nml/actions/action0properties.py

```diff
@@ -255,8 +255,11 @@ def cargo_list(value, max_num_cargos):
     'name'               : {'num': -1, 'string': None},
 }
 
-def ottd_display_speed(value, divisor, unit):
-    return (round(value.value / divisor * 10 * unit.ottd_mul) >> unit.ottd_shift) // 16
+def ottd_display_speed(value, mul, div, unit):
+    # Convert value to km/h-ish.
+    kmh_ish = value.value * mul // div
```

Comment:
While this now matches OTTD's GetDisplaySpeed, it throws away the fractional part of the speed.
You can define a road vehicles that has a max speed of 7 units (=3.5 km-ish/h), and the movement code will use that speed. OTTD will round it for display to 3 km-ish/h though.

So, ideally it should be possible to define vehicles in NML with a speeds of 3, 3.5 and 4 km/h, which have different speed in-game, and the 3 and 4 is also shown in the vehicle lists.

Reply:
This function is only used to clamp the internal value calculated elsewhere, by incrementing/decrementing it until the display speed matches the input. Fractional values that display as the correct integer should be unaffected.